scripts/gpu_support/nvidia/link_nvidia_host_libraries.sh

```diff
@@ -402,8 +402,41 @@ find_cuda_libraries_on_host() {
 
         if [ -n "$matched" ]; then
             log_verbose "Found matches for ${library}: $matched"
-            # Do not quote $matched, since it can contain multiple libraries split by \n!
-            MATCHED_LIBRARIES+=($matched)
+            
+            # Process each matched library and avoid duplicates by filename
+            # Used `while - read <<< $matched`` to handle whitespaces and special characters.
+            while IFS= read -r lib_path; do
+                # Skip empty lines
+                [ -z "$lib_path" ] && continue
+                
+                # Extract just the filename from the path
+                lib_name=$(basename "$lib_path")
+                echo "Checking library $lib_name for duplicates"
+                
+                # Check if we already have this library filename in our matched libraries
+                duplicate_found=0
+                for existing_lib in "${MATCHED_LIBRARIES[@]}"; do
+                    existing_name=$(basename "$existing_lib")
+                    if [ "$existing_name" = "$lib_name" ]; then
+                        log_verbose "Duplicate library found: $lib_name (existing: $existing_lib, new: $lib_path)"
+                        # Prioritize libraries in standard locations if possible
+                        if [[ "$lib_path" == "/usr/lib"* || "$lib_path" == "/lib"* ]]; then
```

Comment:
I'm not sure if this will catch all cases. Why not just take the first option? That is what the linker itself would do.